pkgs/initiator/initiator.go

```diff
@@ -212,68 +212,78 @@ func (c *Initiator) initMessageFlowHandling(init *spec.Init, id [24]byte, operat
 	return finalResults, nil
 }
 
-func (c *Initiator) ResignMessageFlowHandling(rMsg *wire.ResignMessage, id [24]byte, operators []*spec.Operator) ([][]byte, error) {
-	dkgResult, errs, err := c.SendResignMsg(id, rMsg, operators)
+func (c *Initiator) ResignMessageFlowHandling(rMsg *wire.ResignMessage, sig *wire.SignatureForHash, id [24]byte, operators []*spec.Operator) ([][]byte, error) {
+	_, _, err := c.SendResignMsg(id, rMsg, operators)
+	if err != nil {
+		return nil, err
+	}
+	resignResult, errs, err := c.SendSignResignMsg(id, sig, operators)
```

Comment:
This is a bit difficult because it can be one sig for EOA users and multiple sigs (but encoded in one byte array) for multisig users, so its hard to say whether we should use sig or sigs.

Reply:
you always assume a list of signatures
It may be a size 1 list

Reply:
updated